scripts/cnode-helper-scripts/mithril-signer.sh

```diff
@@ -0,0 +1,160 @@
+#shellcheck source=/dev/null
+
+. "$(dirname $0)"/env offline
+
+######################################
+# User Variables - Change as desired #
+# Common variables set in env file   #
+######################################
+
+#MITHRILBIN="${HOME}"/.local/bin/mithril-signer # Path for mithril-signer binary, if not in $PATH
+#HOSTADDR=127.0.0.1                             # Default Listen IP/Hostname for Mithril Signer Server
+MITHRIL_HOME="/opt/mithril/mithril-signer"
+
+######################################
+# Do NOT modify code below           #
+######################################
+
+#####################
+# Functions         #
+#####################
+
+usage() {
+  cat <<-EOF
+		
+		Usage: $(basename "$0") [-d] [-u]
+		
+		Cardano Mithril signer wrapper script !!
+		-d    Deploy mithril-signer as a systemd service
+		-u    Update mithril-signer environment file
+		
+		EOF
+  exit 1
+}
+
+set_defaults() {
+  [[ -z "${MITHRILBIN}" ]] && MITHRILBIN="${HOME}"/.local/bin/mithril-signer
+  [[ -z "${MITHRIL_HOME}" ]] && MITHRIL_HOME=/opt/mithril/mithril-signer
+  if [[ -z "${NETWORK}" ]] || [[ -z "${POOL_NAME}" ]] || [[ "${POOL_NAME}" == "CHANGE_ME" ]]; then
+    echo "ERROR: The NETWORK and POOL_NAME must be set before deploying mithril-signer as a systemd service!!"
+    exit 1
+  else
+    case "${NETWORK}" in
+      mainnet|preprod)
+      RELEASE="release"
+      ;;
+      *)
+      RELEASE="pre-release"
+      ;;
+    esac
+  fi
+  [[ -z ${RELEASE} ]] && echo "ERROR: Failed to set RELEASE variable, please check NETWORK variable in env file!!" && exit 1
+}
+
+pre_startup_sanity() {
+  [[ ! -f "${MITHRILBIN}" ]] && MITHRILBIN="$(command -v mithril-signer)"
+  if [[ ! -S "${CARDANO_NODE_SOCKET_PATH}" ]]; then
+    echo "ERROR: Could not locate socket file at ${CARDANO_NODE_SOCKET_PATH}, the node may not have completed startup !!"
+    exit 1
+  fi
+  # Move logs to archive
+  [[ -f "${LOG_DIR}"/mithril-signer.log ]] && mv "${LOG_DIR}"/mithril-signer.log "${LOG_DIR}"/archive/
+}
+
+generate_environment_file() {
+  ERA_READER_ADDRESS=https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK}/era.addr
+  ERA_READER_VKEY=https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK}/era.vkey
+  sudo bash -c "cat <<-'EOF' > ${MITHRIL_HOME}/service.env
+  AGGREGATOR_ENDPOINT=https://aggregator.${RELEASE}-${NETWORK}.api.mithril.network/aggregator
+  KES_SECRET_KEY_PATH=${POOL_DIR}/${POOL_HOTKEY_SK_FILENAME}
+  OPERATIONAL_CERTIFICATE_PATH=${POOL_DIR}/${POOL_OPCERT_FILENAME}
+  PARTY_ID=$(cat ${POOL_DIR}/${POOL_ID_FILENAME})
+  DB_DIRECTORY=${CNODE_HOME}/db
+  CARDANO_CLI_PATH=${HOME}/.local/bin/cardano-cli
+  DATA_STORES_DIRECTORY=${MITHRIL_HOME}/data-stores
+  ERA_READER_ADAPTER_TYPE=cardano-chain
+  ERA_READER_ADDRESS=https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/${RELEASE}-${NETWORK}/era.addr
```

Comment:
will need exception if we're to stand something for guildnet

Reply:
When I inquired last about guildnet I don't recall hearing anyone was already testing, and didnt get a feeling anyone was interested in doing so. Or at least everyone seemed too busy with other projects at that time.

* If one does exist today point me at it and I'll work out the changes required. If not then I suggest we leave this as is until we have an additional need and layouts to support.
* If anyone is interested in adding mithril to guildnet now that we have tools I would be happy to participate. 
  * If we choose an incompatible structure then **ERA_READER_ADDRESS**, **ERA_READER_VKEY**, and **GENESIS_VERIFICATION_KEY** can be migrated to the `set_defaults()` where choices are made based on the NETWORK.